### PR TITLE
extend vit and add mae model and finetune checkpoint file

### PR DESCRIPTION
**Pull request by sageyou**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

1) rebase pr https://github.com/mindspore-lab/mindcv/pull/733 of SamitHuang
2) update model vit
3) add and update model mae (also with chheckpoint file fo mae finetune)
4) remaining tasks： upload new checkpoint files of vit （after being merged of this pr）

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs
https://github.com/mindspore-lab/mindcv/pull/733
(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Comment by vigo999** [comment]
@SamitHuang @geniuspatrick please review

File: mindcv/models/helpers.py [hunk 1]
```diff
@@ -198,3 +202,105 @@ def build_model_with_cfg(
             raise RuntimeError(f"`feature_only` is not implemented for `{model_cls.__name__}` model.") from e
 
     return model
+
+
+def interpolate_relative_position_bias(checkpoint_params, network):
```
Comment (geniuspatrick) [inline mindcv/models/helpers.py:207]: 下面这俩函数好像没有用到？

**Review by geniuspatrick — APPROVED** [review]

**Comment by vigo999** [comment]
原本的vit 是否可以提供vit encoder? 不单独搞vit_encoder.py @sageyou @geniuspatrick

**Comment by sageyou** [comment]
> 原本的vit 是否可以提供vit encoder? 不单独搞vit_encoder.py @sageyou @geniuspatrick

mae可以实现，但是会麻烦 结构都要动。
beit不行，因为新的vit_encoder就是出于mae和beit可以共同使用enocoder而写的，里面增加了新的功能供beit使用, 例如Attention中可以使用relative_position_bias，encoder中可选择使用LayerScale等，具体请看治锋之前提的一个issue:https://github.com/mindspore-lab/mindcv/issues/693。
总的来说新的可以兼容原来的vit encoder， 非要整改的话， 个人建议改原来的vit。

**Comment by sageyou** [comment]
review plz, @SamitHuang @geniuspatrick

**Comment by geniuspatrick** [comment]
LGTM. BTW, have the new weights been verified?

**Comment by sageyou** [comment]
> LGTM. BTW, have the new weights been verified?

@geniuspatrick Yes, I have validated all of them, including vit and mae.
However,  the ckpt file of vit has not been uploaded yet: the url of pretrained-vit of this pr is not the latest.  I would like to do it next~

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**